### PR TITLE
Fix Battles using all units instead of selected units

### PR DESCRIPTION
**Pull request by lotuuu**
Closes https://github.com/orgs/lambdaclass/projects/18/views/7?pane=issue&itemId=56587539

Battle was using all units instead of only selected units. This PR fixes that.

To check if this works:
0- Add an inspect in the code to print the units that are being used in fights. This should be an `IO.inspect(team_1) `call at the top of the `battle` function in `Champions.Battle`

1- Create a user
{:ok, user} = Champions.Users.register("Name")

2- Unselect a unit (get the id from user.units)
Champions.Units.unselect_unit(user.id, unit_id)

3- Run a battle against a level (check #337 for tutorial). Check that the unselected unit is not on the printed team.

### CODE REVIEW

**Review by agustinesco — APPROVED** [review]
LGTM!

File: apps/champions/lib/champions/battle.ex [hunk 1]
```diff
@@ -7,19 +7,22 @@ defmodule Champions.Battle do
...
+  alias GameBackend.Units
   alias GameBackend.Users
 
   @doc """
   Plays a level for a user, which means fighting its units with their selected ones.
   Returns `:win` or `:loss` accordingly, and updates the user's progress if they win..
   """
   def fight_level(user_id, level_id) do
-    with {:user, {:ok, user}} <- {:user, Users.get_user(user_id)},
+    with {:user, {:ok, _user}} <- {:user, Users.get_user(user_id)},
```
Comment (ncontinanza) [inline apps/champions/lib/champions/battle.ex:18]: What's the reason of this `get_user()` here? Just checking that it exists?

**Review by ncontinanza — COMMENTED** [review]
Looking good! I left a question to understand better one of the changes

File: apps/champions/lib/champions/battle.ex (hunk 1 above)
Comment (lotuuu) [reply apps/champions/lib/champions/battle.ex:18]: yes!

Comment (lotuuu) [reply apps/champions/lib/champions/battle.ex:18]: But we have a new User.exists? function now so i'm going to replace this with that one

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Review by agustinesco — APPROVED** [review]
LGTM!

**Pull request merged**